.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,66 @@
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: End to End test
+
+on:
+  push:
+  # Run on branch/tag creation
+  create:
+  # Run on pull requests
+  pull_request:
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    # strategy:
+    #   matrix:
+    #     python-version: ["3.10"]
+    steps:
+      # - name: Set up Python ${{ matrix.python-version }}
```

Comment:
@FrancescoVaiani 
Unfortunately, this action seems to be somehow interfering with the `astarte-cluster-action@v1` action.
If I enable this action all HTTPS requests performed with the `requests` module fail. The returned error can be seen [here](https://github.com/sorru94/astarte-device-sdk-python/actions/runs/4202699226/jobs/7291185267).
I think the issue is somehow related to the path to the trusted certificates that might be changed by this action.
As a quick and dirty fix, I disabled this action and called directly `python3.10` down below.
This works as long as the Ubuntu distro used by the action natively includes Python 3.10. This is the case right now as the latest image of ubuntu included in the pipeline is 22.04. See [here](https://github.com/actions/runner-images).

Reply:
I suggest using python 3.8 as a target. 
Its lifetime is in a year and a half and it is still widely used.
In addition, later versions are backward compatible and we should be moderately confident that they will work.
Remember to remove the commented lines.